tests/integration/deploy/test_deploy_command.py

```diff
@@ -745,6 +746,30 @@ def test_deploy_with_no_s3_bucket_set_resolve_s3(self, template_file):
         deploy_process_execute = run_command(deploy_command_list)
         self.assertEqual(deploy_process_execute.process.returncode, 0)
 
+    @parameterized.expand(["aws-serverless-function.yaml"])
+    def test_deploy_with_no_s3_bucket_set_resolve_s3_clean_state(self, template_file):
```

Comment:
Can we make this a separate class instead? I would want us to have a separate class and file for testing that creation of the managed stack works as expected.